### PR TITLE
#81 EA-2188 implemented custom filtering for total field

### PR DESCRIPTION
**Pull request by rgraf**

### CODE REVIEW

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 1]
```diff
@@ -119,7 +125,8 @@
 			WebUserSetFields.MAX_ITEMS_PER_PAGE, profile);
 	    }
 
-	    String serializedUserSetJsonLdStr = serializeUserSet(profile, storedUserSet);
+	    UserSet resUserSet = setTotalField(profile, storedUserSet);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:128]: acording to the desing this functionality must be implemented in the  applyProfile method which is already called in serializeUserSet method.
UserSet set = getUserSetService().applyProfile(storedUserSet, profile);

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 2]
```diff
@@ -200,7 +207,8 @@
 		userSet = getUserSetService().fetchItems(userSet, sort, sortOrder, pageNr, pageSize, profile);
 	    }
 
-	    String userSetJsonLdStr = serializeUserSet(profile, userSet);
+	    UserSet resUserSet = setTotalField(profile, userSet);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:210]: move to applyProfile method as indicated in previous comment

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 3]
```diff
@@ -457,7 +466,8 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
 	    checkIfMatchHeader(eTagOrigin, request);
 
 	    UserSet updatedUserSet = getUserSetService().insertItem(datasetId, localId, position, existingUserSet);
-	    String serializedUserSetJsonLdStr = serializeUserSet(profile, updatedUserSet);
+	    UserSet resUserSet = setTotalField(profile, updatedUserSet);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:469]: move to applyProfile method as indicated in previous comment

File: set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java [hunk 4]
```diff
@@ -200,4 +199,18 @@ protected String serializeUserSet(LdProfiles profile, UserSet storedUserSet) thr
...
         return getConfiguration().getApiVersion();
     }
+    
+    /**
+     * For the open sets with minimal profile we set the value to -1 so that the total will not be serialized
+     * @param profile
+     * @param userSet
+     * @return user set with updated total field
+     */
+    protected UserSet setTotalField(LdProfiles profile, UserSet userSet) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java:209]: method not needed, please inline the code in applyProfile method

**Review by gsergiu — CHANGES_REQUESTED** [review]
please implement required changes

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 1 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:128]: functionality moved to applyProfile method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 2 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:210]: in all similar methods, functionality moved to applyProfile method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 3 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:469]: functionality moved to applyProfile method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java (hunk 4 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java:209]: code of this method integrated in applyProfile method

**Review by gsergiu — APPROVED** [review]

**Pull request merged**